src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ResolvedReferenceHandling.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    //Consider adding WriteReference value to use that instead of output param writeAsReference in ReferenceHandlingStrategy.
+    internal enum ResolvedReferenceHandling
```

Comment:
The reasoning for the name I choose was that, even though you specify a `ReferenceHandling` behavior, this will vary depending on the object/property you are writing. So we need to first **resolve** what is going to be written at the start of the object.

I am just trying to justify my naming, if you think it is more appropriate your suggestion, I am open to switch.